src/_common/events/subscription.ts

```diff
@@ -0,0 +1,26 @@
+export type UnsubscribeFunc = () => void;
+
+export class Subscription {
+	private unsubFunc: UnsubscribeFunc | null;
+
+	constructor(private subscriber: () => UnsubscribeFunc) {
```

Comment:
This unsub logic twisted my head into a pretzel

Reply:
i like pretzels.
anyways the idea is that you can resubscribe to a topic you've previously unsubscribed to. also this way you can't just randomly create a subscription for a topic without going through topic. i hate having many ways to do the same thing.

Reply:
I think we need some sort of comment on the EventSubscription or the ctor to explain what's going on here. I can figure it out, but it's kinda convoluted and I'm sure when I actually want to use this, I'll have to go figure it out again.
Some simple explanation of what a subscriber is, what an UnsubscribeFunc is and when/why to use it would be gud.

Reply:
I wouldn't say its convoluted, considering thats just the standard naming convention for the components in pubsub systems. I don't mind adding comments tho.

Reply:
removed the resubscribing bit, and actually the entire event subscription. its just a ts type now.